tests/notebooks/test_run_notebooks.py

```diff
@@ -1,54 +1,295 @@
 import os
 
+import nbformat
 import pytest
 from pyomo.common.fileutils import this_file_dir
 from testbook import testbook
 
 from omlt.dependencies import keras_available, onnx_available
 
+# TODO: We need to try and write these tests to rely on internal consistencies and less on absolute numbers and tolerances
 
-# TODO: These will be replaced with stronger tests using testbook soon
-def _test_run_notebook(folder, notebook_fname, n_cells):
-    # change to notebook directory to allow testing
-    cwd = os.getcwd()
+
+# return testbook for given notebook
+def open_book(folder, notebook_fname, **kwargs):
+    execute = kwargs.get("execute", True)
     os.chdir(os.path.join(this_file_dir(), "..", "..", "docs", "notebooks", folder))
-    with testbook(notebook_fname, timeout=300, execute=True) as tb:
-        assert tb.code_cells_executed == n_cells
-    os.chdir(cwd)
+    book = testbook(notebook_fname, execute=execute, timeout=300)
+    return book
+
+
+# checks that the number of executed cells matches the expected
+def check_cell_execution(tb, notebook_fname, **kwargs):
+    injections = kwargs.get("injections", 0)
+    assert (
+        tb.code_cells_executed
+        == cell_counter(notebook_fname, only_code_cells=True) + injections
+    )
+
+
+# checks for correct type and number of layers in a model
+def check_layers(tb, activations, network):

```

Comment:
It might be better if the function names match more what the function does (rather than what it is used for). This function actually injects code into the notebook. Maybe "inject_activation_check"